templates/searchbox.html

```diff
@@ -7,28 +7,26 @@
             class="form-control"
             id="ticker"
             placeholder="Ticker Symbol"
-            maxlength=5
-            onkeydown="return validateInput(event)"/>
+            maxlength=5/>
       </div>
       <div class="col-md-1 nopadding">
-        <button
+        <input
             type="submit"
+            value="Analyze"
             class="btn btn-raised btn-md btn-warning"
-        >Analyze</button>
+        >
       </div>
     </div>
   </form>
 </div>
 
 <script>
-  // Returns `true` if the key is an alphabetic keystroke or the backspace
-  // key.
-  function validateInput(event) {
-    let key = event.keyCode;
-    const backspaceKey = 8;
-    const periodKey = 190;
-    return ((key >= 65 && key <= 90) || key == backspaceKey || key == periodKey);
-  };
+document.querySelector("#ticker").addEventListener("keypress", (event) => {
+  let key = event.key;
+  if (!key.match(/[A-Za-z]/)) {
```

Comment:
Doesn't this also need to be able to match periods? For tickers like `BRK.B`?

(We have some inconsistencies now. Ideally we'd also accept formats like `BRK-B`. But some of the backends are particular about which one they use. And we don't currently sanitize the input for each backend to make sure it's in the accepted format. That's a separate bug but one we can also fix up to improve things)

Reply:
Fixed in the latest version. Thanks!